woocommerce/google-listings-and-ads
# Add extra product attributes

### Changes proposed in this Pull Request:

This PR adds some of the extra product attributes like GTIN, MPN, etc. when syncing them to Google Merchant Center. It adds a new tab to the product edit page and allows the custom attributes to be entered by the user via a form. It also checks for integrations by providing filters to modify the attribute's "acceptable values" and the actual value.

Part of #16.

### Testing instructions
1. Create a new product or edit an existing one
2. There should be a new tab called "Google listings and ads" on the product data section of the edit product page
3. There should be a field to enter MPN and GTIN for the product
4. You can select one of the following for these attributes
        - `Default`: This means that the value will be set to the default value specified on a global/category level (not yet implemented)
        - __Values from integrations__: For now we have only one integration for the Yoast WooCommerce Seo extension, so if you have that installed there will be a value called "From Yoast WooCommerce SEO" displayed there
        - `Enter a custom value`: If this is selected a new text box will be displayed under the current attribute to allow the user to enter their own custom value for this product
5. Choose any of the values and update/publish the product
6. Wait until the product gets synced to Google (you can check the action scheduler job or just sync it manually via the Connection Test page)
7. Check the synced product details on Google Merchant Center and make sure that the attributes you set for them are visible in there
8. Repeat the testing steps for variable products and this time set the attributes per each variation

### Additional information
- We do a simple validation on the attributes on our end. For example, GTIN length is validated but we don't validate its correctness (e.g. check digit validation).
- To test the Yoast WooCommerce SEO integration without installing it (since it's not free), you could set a global constant in `wp-config` called `WPSEO_WOO_VERSION` (with any value) and then create a new metadata for your product in the `wp_postmeta` table with the key `wpseo_global_identifier_values` and the following value as an example:
```
a:6:{s:5:"gtin8";s:8:"12345678";s:6:"gtin12";s:0:"";s:6:"gtin13";s:0:"";s:6:"gtin14";s:0:"";s:4:"isbn";s:0:"";s:3:"mpn";s:9:"123456789";}
```
